fotoapparat/src/main/java/io/fotoapparat/hardware/v1/RawCameraParametersProvider.java

```diff
@@ -0,0 +1,82 @@
+package io.fotoapparat.hardware.v1;
+
+import android.hardware.Camera;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Provides {@link Camera.Parameters} with methods to extract additional camera parameters.
+ */
+@SuppressWarnings("deprecation")
+public class RawCameraParametersProvider {
+
+    /* Raw camera params keys */
+    private static final String[] RAW_ISO_SUPPORTED_VALUES_KEYS = {
+            "iso-values", "iso-mode-values", "iso-speed-values", "nv-picture-iso-values"
+    };
+    private static final String[] RAW_ISO_CURRENT_VALUE_KEYS = {
+            "iso", "iso-speed", "nv-picture-iso"
+    };
+
+    private Camera.Parameters cameraParameters;
+
+    public RawCameraParametersProvider(Camera.Parameters cameraParameters) {
+        this.cameraParameters = cameraParameters;
+    }
+
+    public Camera.Parameters getCameraParameters() {
+        return cameraParameters;
+    }
+
+    /**
+     * Returns set of ISO values, that camera supports.
+     *
+     * @return the set of supported ISO values.
+     */
+    @NonNull
+    public Set<Integer> getSensorSensitivityValues() {
+        final Set<Integer> isoValuesSet = new HashSet<>();
+
+        String[] rawValues = extractRawCameraValues(RAW_ISO_SUPPORTED_VALUES_KEYS);
+
+        // Should log that result is nullable or not?
+        if (rawValues != null) {
+            for (String value : rawValues) {
+                try {
+                    isoValuesSet.add(Integer.valueOf(value.trim()));
+                } catch (NumberFormatException e) {
+                    // Found not number option. Skip it.
+                }
+            }
+        }
+
+        return isoValuesSet;
+    }
+
+    @Nullable
+    private String[] extractRawCameraValues(@NonNull String[] keys) {
+        String[] rawValues = null;
+
+        for (String key: keys) {
+            rawValues  = extractRawCameraValues(key);
+            if (rawValues != null) {
+                break;
+            }
+        }
+
+        return rawValues;
+    }
+
+    @Nullable
+    private String[] extractRawCameraValues(@NonNull String key) {
```

Comment:
Instead of `String[]` it's more solid to have a class KeyValue to return the pair or null is they don't exist

Reply:
Isn't it overkill for just getting raw string values from `cameraParameters`? Especially if `KeyValue` class, in my opinion, should have method `<T> T getValue(Class<T> clazz)`, that will be implemented with different checks on what returned value should be and then casting raw string to fit this class.

Reply:
Ah sorry, I think I didn't explain it very good and I just took a second look. This has been build with the purpose of reusuability, I now see 👍 . 

What I mainly think it's not nice is the `NumberFormatException` on a upper/later level.

So we can just make a nice logic only focusing to extract the ending `Set<Integer>` of ISOs and when we bump on the issue of duplicate/reusable code later on (because some similar behaviour with iso and splitting) we can think & solve it at that point.